### PR TITLE
added Book DTO & mapper classes, BookController class

### PR DESCRIPTION
**Pull request by DanilKi**
added findById() method to BookRepository & BookService

### CODE REVIEW

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/controller/BookController.java]: If you have only one constructor, don't need to use `@Autowired`, just use `@RequiredArgsConstructor`

File: src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -1,28 +1,45 @@
...
 import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java]: The same as in controller, use `@RequiredArgsConstructor`

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's improve your solution

**Review by sarakhmen — APPROVED** [review]
Good job!

**Pull request merged**